docs: add DEP introduction (#648)
medibloc/panacea-core: .gitbook/0-about-panacea/1-data-exchange-protocol/1-DEP-problems-it-solves.md

# DEP and the problems it solves

DEP seeks to address the problems of decentralized data delivery that allows for data validation while ensuring data privacy:

- [Data privacy](#data-privacy)
- [Data validation](#data-validation)
- [Incentive](#incentive)
- [Decentralization](#decentralization)

> - [Scalability](#scalability)

Reviewer: For instance, https://vac.dev/principles

> - [Scalability](#scalability)

Author: Good I tried to use `principles` at [d29532f](https://github.com/medibloc/panacea-core/pull/648/commits/d29532f7084c092ba1d7d0d8b3f8bf792370c92d).

## Data privacy

Since DEP is targeting sensitive data, the privacy of the data must be ensured throughout all data transfers. 

> It is essential that no data is exposed in the process of data delivery.

Reviewer: ```suggestion
No data must be exposed to anyone except promised data consumers during the data delivery.
```

> DEP uses data encryption in all data delivery processes to ensure the privacy of sensitive data. 

Reviewer: ```suggestion
DEP uses data encryption in the whole data delivery process to ensure the privacy of sensitive data. 
```

> Symmetric and asymmetric key encryption is utilized to ensure that the privacy of data during all data delivery processes.

Reviewer: ```suggestion
ECIES hybrid encryption is used to ensure that the privacy of data during the entire data delivery process.
```
https://cryptobook.nakov.com/asymmetric-key-ciphers/ecies-public-key-encryption

## Data validation

> As a data consumer, it is important to validate that the data you receive is the right data.

Reviewer: ```suggestion
As data consumers, it is crucial to validate that the data they receive is correct.
```

> The data must be truthful, uncorrupted, and conform to the requirements the consumer propose.

Reviewer: ```suggestion
The data must be truthful, untampered, and conform to the requirements the consumer specified.
```

> DEP provides a way to verify data by utilizing oracle running in an enclave environment, did-based verifiable presentation.

Reviewer: ```suggestion
DEP provides a way to verify data by utilizing oracle running in an secure enclave environment and DID-based verifiable presentation.
```

## Incentive

> By providing data, the data provider can be guaranteed with an incentive offered by the data consumer. 
> In this case, the delivery of data and the paying incentive must be processed atomically and simultaneously. 

Reviewer: ```suggestion
By providing data, the data provider can be guaranteed an incentive paid by the data consumer. 
In this case, data delivery and payment must be processed atomically and simultaneously.
```

> DEP ensures this through the panacea blockchain Tx and data encryption.

Reviewer: ```suggestion
DEP ensures this through Panacea blockchain transactions and data encryption.
```

## Decentralization

> DEP aims to be a decentralized protocol with no centralized data validator and deliverer.

Reviewer: ```suggestion
DEP aims to be a decentralized protocol with no centralized mediator for data validation and delivery.
```

> The DEP components, panacea blockchain and oracle, are open decentralized ecosystem where everyone can participate.

Author: I think this part is quite weak. If you have any suggestions, please comments.

Reviewer: ```suggestion
## Decentralization

DEP aims to be a decentralized protocol with no centralized data validator and deliverer.

Anyone can participate in operating DEP components, such as Panacea blockchain and oracle. By the nature of the proof-of-stake blockchain, any party with less than 2/3 voting power cannot change the behavior of DEP, and anyone with less than 1/3 voting power cannot stop the protocol. Also, anyone can operate oracle nodes to improve the performance of DEP and earn incentives if they have appropriate hardware environments.
The entire source code, protocol specifications, and discussions are open-sourced, so anyone can dive into details and participate in developments. Any protocol changes must go through governments to be deployed in production. 
```

Author: Thank you. I think this is clearer.

- [Data privacy](#data-privacy)
- [Data validation](#data-validation)
- [Incentive](#incentive)
- [Decentralization](#decentralization)
- [Scalability](#scalability)

## Data privacy

Since DEP is targeting sensitive data, the privacy of the data must be ensured throughout all data transfers. 
No data must be exposed to anyone except promised data consumers during the data delivery.

DEP uses data encryption in the whole data delivery process to ensure the privacy of sensitive data. 
Symmetric and asymmetric key encryption is utilized to ensure that the privacy of data during all data delivery processes.

## Data validation

As data consumers, it is crucial to validate that the data they receive is correct.
The data must be truthful, untampered, and conform to the requirements the consumer specified.

DEP provides a way to verify data by utilizing oracle running in an secure enclave environment and DID-based verifiable presentation.

## Incentive

By providing data, the data provider can be guaranteed an incentive paid by the data consumer. 
In this case, data delivery and payment must be processed atomically and simultaneously.

> DEP ensures this through Panacea blockchain transactions and data encryption.

Author: I wanted to talk about the problem that incentive payments and data delivery should be handled atomically. Our DEP is designed so that consumers can only see the data if the incentives are paid.

I think the confusion comes from the title `Incentive`. I'm not sure if `Atomic Incentive` would be better, so if you have any thoughts, please comment!

> DEP ensures this through Panacea blockchain transactions and data encryption.

Reviewer 2: Aha. It's not problem.
I understand it.

- [Scalability](#scalability)

## Data privacy

Since DEP is targeting sensitive data, the privacy of the data must be ensured throughout all data transfers. 
No data must be exposed to anyone except promised data consumers during the data delivery.

DEP uses data encryption in the whole data delivery process to ensure the privacy of sensitive data.
[ECIES hybrid encryption](https://cryptobook.nakov.com/asymmetric-key-ciphers/ecies-public-key-encryption) is used to ensure data privacy during the entire data delivery process.

## Data validation

As data consumers, it is crucial to validate that the data they receive is correct.
The data must be truthful, untampered, and conform to the requirements the consumer specified.

DEP provides a way to verify data by utilizing oracle running in an secure enclave environment and DID-based verifiable presentation.

## Incentive

By providing data, the data provider can be guaranteed an incentive paid by the data consumer. 
In this case, data delivery and payment must be processed atomically and simultaneously.

DEP ensures this through Panacea blockchain transactions and data encryption.

## Decentralization

> DEP aims to be a decentralized protocol with no centralized mediator for data validation and delivery.

Reviewer 3: Does we need to say `with no centralized mediator`?

Reviewer: could you please elaborate or suggest other options? i also think this sentence can be better but also it's okay as it is for now.

Reviewer 3: I just feel that the `decentralized protocol with no centralized mediator` little awkward. But for now, I think it is okay to go.

Reviewer: Man, if you feel it's awkward, please give us your suggestion. If not, the PR author doesn't understand what you mean.

Reviewer: I guess you feel it's awkward to use `mediator` together with `centralized`

Reviewer 3: Ah, I mean that it's awkward to use `decentralized protocol` together with `with no centralized mediator`. Because everybody knows that decentralized means that already didn't have any mediator.
So, I want to suggest like this:
`DEP aims to be a decentralized protocol for data validation and delivery.`
Sorry for the not detailed comment.

Author: I think it's more of a statement of emphasis. It's not partially decentralized, it's fully decentralized in the validation and delivery process.

Let's think of some other ways to express it.

Reviewer 3: Ah, if you are using the expression for emphasis, it is not awkward.

Reviewer: i agree with opinions from both of you. 
i'm okay with the emphasis that MS used. IMHO, i'm not trying to touch expression details at this time unless the meaning of the sentence is unclear.
